Framework/MDAlgorithms/test/IntegratePeaksMD2Test.h

```diff
@@ -180,6 +180,128 @@ class IntegratePeaksMD2Test : public CxxTest::TestSuite {
+            3.0); // 27 times the volume / 9 times the counts = 1/3 density
+
+    // --- Make a fake PeaksWorkspace ---
+    PeaksWorkspace_sptr peakWS(new PeaksWorkspace());
+    Instrument_sptr inst = ComponentCreationHelper::createTestInstrumentCylindrical(5);
+    peakWS->addPeak(Peak(inst, 1, 1.0, V3D(0., 0., 0.)));
+    TS_ASSERT_EQUALS(peakWS->getPeak(0).getIntensity(), 0.0);
+    AnalysisDataService::Instance().addOrReplace("IntegratePeaksMD2Test_peaks", peakWS);
+
+    std::vector<double> peakRadius(numberValuesPeakRadius, 1.0);
+    std::vector<double> bkgInnerRadius(numberValuesBkgInnerRadius, 1.1);
+    std::vector<double> bkgOuterRadius(numberValuesBkgOuterRadius, 2.0);
+
+    IntegratePeaksMD2 alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("InputWorkspace", "IntegratePeaksMD2Test_MDEWS"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("PeakRadius", peakRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("PeaksWorkspace", "IntegratePeaksMD2Test_peaks"));
+    TS_ASSERT_THROWS_NOTHING(alg.setPropertyValue("OutputWorkspace", "IntegratePeaksMD2Test_Output"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("BackgroundInnerRadius", bkgInnerRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("BackgroundOuterRadius", bkgOuterRadius));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Ellipsoid", ellip));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Cylinder", cyl));
+    if (shouldPass) {
+      TS_ASSERT_THROWS_NOTHING(alg.execute());
+    } else {
+      TS_ASSERT_THROWS(alg.execute(), const std::runtime_error &);
+    }
+    AnalysisDataService::Instance().remove("IntegratePeaksMD2Test_peaks");
+  }
+  /** In the following validation tests, three validation principles must be met:
+      (where radius can be peak/BackgroundInner/BackgroundOuter)
+   A) radius vector has 1 or 3 values
+   B) radius vector has 1 value for sphere/cylinder/ellipsoid, 3 values only for ellipsoid
+      (sphere implied when ellipsoid=false and cylinder=false)
+   C) ellipsoid and cylinder cannot both be true
+   */
+  /// Peak Radius
+  void test_validationPeak1ValueSphere() { simpleRun(true, 1, 1, 1, false, false); }
+  void test_validationPeak1ValueCylinder() { simpleRun(true, 1, 1, 1, true, false); }
+  void test_validationPeak1ValueEllipsoid() { simpleRun(true, 1, 3, 3, false, true); }
+  void test_validationPeak1ValueEllipsoidAndCylinder() { simpleRun(false, 1, 1, 1, true, true); } // fail C
+
+  void test_validationPeak2ValuesSphere() { simpleRun(false, 2, 1, 1, false, false); }             // fail A
+  void test_validationPeak2ValuesCylinder() { simpleRun(false, 2, 1, 1, true, false); }            // fail A
+  void test_validationPeak2ValuesEllipsoid() { simpleRun(false, 2, 3, 3, false, true); }           // fail A
+  void test_validationPeak2ValuesEllipsoidAndCylinder() { simpleRun(false, 2, 1, 1, true, true); } // fail A + C
```

Comment:
I think once we've checked C (ellip and cyl can't both be true) with valid radii (as on L239) we don't need to check again (with various invalid combinations of radii?). For example on this line (L244).

Reply:
I can imagine a bug where say inputting 3 radii values means that Ellipsoid and Cylinder runs fine, where as inputting other numbers of radii values is correctly caught, so I'd like to keep it!

Reply:
My understanding is that  both ellipsoid and cylinder being true should cause an exception for all radii (valid or invalid) - to test this only requires running once with valid radii. The validity of various number of radii passed to ellipsoid and cylinder are tested elsewhere.